src/SDKs/Search/DataPlane/Microsoft.Azure.Search.Data/Customizations/Documents/DocumentsOperations.cs

```diff
@@ -123,97 +119,160 @@ internal DocumentsOperations(SearchIndexClient client)
             };
         }
 
-        public Task<AzureOperationResponse<DocumentSearchResult>> ContinueSearchWithHttpMessagesAsync(
+        public async Task<AzureOperationResponse<DocumentSearchResult>> ContinueSearchWithHttpMessagesAsync(
             SearchContinuationToken continuationToken,
             SearchRequestOptions searchRequestOptions = default(SearchRequestOptions),
             Dictionary<string, List<string>> customHeaders = null,
             CancellationToken cancellationToken = default(CancellationToken))
         {
-            string invocationId;
-            Guid? clientRequestId;
-            bool shouldTrace =
-                ValidateAndTraceContinueSearch(
-                    continuationToken,
+            JsonSerializerSettings deserializationSettings =
+                JsonUtility.CreateDocumentDeserializerSettings(this.Client.DeserializationSettings);
+
+            bool useGet = continuationToken.NextPageParameters == null;
+
+            AzureOperationResponse<DocumentSearchResultProxy> response;
+
+            if (useGet)
+            {
+                var parameters = ParseQueryParameters(continuationToken.NextLink);
+                string searchText = parameters.ContainsKey("search") ? parameters["search"].First() : null;
+
+                SearchParameters searchParameters = SearchParameters.FromDictionary(parameters);
+
+                response = await this.Client.DocumentsProxy.SearchGetWithHttpMessagesAsync(searchText, Accept, searchParameters,
+                    searchRequestOptions,
+                    customHeaders,
+                    cancellationToken,
+                    requestDeserializerSettings: deserializationSettings).ConfigureAwait(false);
+            }
+            else
+            {
+                response = await this.Client.DocumentsProxy.SearchPostWithHttpMessagesAsync(
```

Comment:
There is a risk in this approach to implementing `ContinueSearch`, although it's a small one. The REST API can actually return some query string parameters even for POST, and technically these should be sent back in the next POST request. Fortunately, the only such query string parameter today is an undocumented internal parameter that the SDK does not use, but if this were ever to change for some reason, this implementation would no longer work.

I think the takeaway here for us is that we should make sure that all new publicly-documented parameters for Search POST should go in the request body, except for `api-version`.